R/DBH_api_tabell.R

```diff
@@ -24,65 +24,64 @@
-    exclude = NULL,
-    variables=list()) {
+    group_by,
+    sort_by,
+    exclude,
+    variables) {
 
     filter_query <- list()
-
     exclude_warning <- character(0)
+    .as_character_utf8 <- function(s) enc2utf8(as.character(s))
 
     for (i in seq_along(filters)) {
       filter_query[[i]] <-
         list(variabel = enc2utf8(names(filters)[i]),
              selection = list(filter = "item",
                               values = lapply(filters[[i]],
-                                      function(s) enc2utf8(as.character(s)))))
+                                              .as_character_utf8)))
       if (filters[[i]][1] == "*") {
         filter_query[[i]]$selection$filter <- "all"
-
-      }
-      else if (filters[[i]][1] %in% c("top",  "lessthan", "greaterthan")) {
+      } else if (filters[[i]][1] %in% c("top",  "lessthan", "greaterthan")) {
         filter_query[[i]]$selection$filter <- filters[[i]][1]
         filter_query[[i]]$selection$values <- lapply(filters[[i]][2],
-                                  function(s) enc2utf8(as.character(s)))
-
+                                                     .as_character_utf8)
       } else if (filters[[i]][1] == "between") {
         filter_query[[i]]$selection$filter <- "between"
-        filter_query[[i]]$selection$values <- c(lapply(filters[[i]][2], function(s) enc2utf8(as.character(s))),
-                                       lapply(filters[[i]][3], function(s) enc2utf8(as.character(s))))
-
+        filter_query[[i]]$selection$values <- lapply(filters[[i]][2:3], .as_character_utf8)
       }
       if (names(filters)[i] %in% names(exclude)) {
-        filter_query[[i]]$selection$exclude <-lapply(exclude[[names(filters)[i]]],
-                                         function(s) enc2utf8(as.character(s)))
+        if (filter_query[[i]]$selection$filter %in% c("all", "top", "lessthan", "greaterthan", "between")) {
+          filter_query[[i]]$selection$exclude <- lapply(exclude[[names(filters)[i]]],
+                                                        .as_character_utf8)
+        } else {
+          exclude_warning <- c(exclude_warning, names(filters)[i])
+        }
       }
-      else if (names(filters)[i] %in% names(exclude)) {
-        exclude_warning <-
-          paste0(ifelse(length(exclude_warning) == 0,
-                        "Excluding filters cannot be combined with the item filter for the following variables: ",
-                        paste0(exclude_warning, ", ")),
-                 names(filters)[i])
-      }
-
     }
-    if(is.null(group_by))
-    {
-      group_by=.dbh_groupBy(table_id)
 
+    if (is.null(filters)) {
+      metadata <- .get_metadata(table_id)
+      char_variabel <- metadata[metadata$Datatype %in% "char", ][["Variabel navn"]][1]
+      filter_query <- list(list(variabel = char_variabel,
+                                selection = list(filter = "all",
+                                                 values = list("*"))))
```

Comment:
Bruker bare en vilkårlig variabel med filter 'all', hvis det ikke er oppgitt noe (av type char for å unngå summeringsvariabler som ikke kan brukes i filter)